### DISCUSSION TITLE
Enforce UTC datetime?

### DISCUSSION THREAD

**Discussion by oerp-odoo** [fastapi/fastapi#12094] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from pydantic import BaseModel, NaiveDatetime

class Sale(BaseModel):
    order_date: NaiveDatetime
    commitment_date: datetime.datetime
```

### Description

I want to enforce datetime that would accept UTC only, without timezones. Because then I won't need to deal with timezones (as in my case datetime is stored in UTC time).

I thought it is possible to use `NaiveDatetime` from pydantic, but when I look at fastapi generated docs, showed format is the same, making no difference whether I use NaiveDatetime or standard datetime.datetime:

```json
{
  "order_date": "2024-08-31T08:04:01.946Z",
  "commitment_date": "2024-08-31T08:04:01.946Z",
  "client_order_ref": "string",
  "lines": [
    {
      "sku": "string",
      "quantity": 0
    }
  ]
}
```

### Operating System

Linux

### FastAPI Version

0.112.2

### Pydantic Version

2.8.2

### Python Version

3.12.3

**Comment by YuriiMotov** [comment, marked as answer]
It's only example value. In fact FastAPI will show 422 error if you try to pass datetime with timezone to `order_date`.

It's Swagger that generates this example. You can specify correct example manually:
```
    order_date: NaiveDatetime = Field(examples=[str(datetime.datetime.now())[:-1]])
```
